Unix/wsman/wsmanclient.c

```diff
@@ -54,15 +55,34 @@ struct _WsmanClient
     EnumerationState *enumerationState;
 };
 
-static void PostResult(WsmanClient *self, const MI_Char *message, MI_Result result)
+static void PostResult(WsmanClient *self, const MI_Char *message, MI_Result result, const MI_Instance *cimError)
 {
     if (Atomic_CompareAndSwap(&self->sentResponse, (ptrdiff_t)MI_FALSE, (ptrdiff_t)MI_TRUE) 
         == (ptrdiff_t)MI_FALSE)
     {
+        OMI_Error *omiError = (OMI_Error*)cimError;
+        OMI_Error *newError;
+        MI_Value value;
+
         PostResultMsg *errorMsg = PostResultMsg_New(0);
 
-        errorMsg->errorMessage = message;
+        if (message)
+        {
+            errorMsg->errorMessage = Batch_Tcsdup(errorMsg->base.batch, message);
+        }
+        else
+        {
+            errorMsg->errorMessage = NULL;
+        }
         errorMsg->result = result;
+        if (omiError && omiError->ProbableCause.exists)
+        {
+            OMI_ErrorFromErrorCode(errorMsg->base.batch, MI_RESULT_TIME_OUT, MI_T("MI"), MI_T("Timedout"), &newError);
```

Comment:
We are going to support more than one type of probable cause so I would not hard code this to timeout.
Also, the error type should be MI_RESULT_TYPE_MI.
We can generate the omi_error with strings in them such that we can add the ProbableCauseDecription automatically from the probable cause.
MI_RESULT_TIME_OUT is not a user facing error which means if that is the error code that goes into the mi result value then the user has no clue what it is. It really is an MI_RESULT_FAILED, unfortunately.